docs/data/data-grid/filtering/CustomHeaderFilterOperatorDataGridPro.js

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Box from '@mui/material/Box';
+import Rating from '@mui/material/Rating';
+import { DataGridPro, getGridNumericOperators } from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VISIBLE_FIELDS = ['name', 'rating', 'country', 'dateCreated', 'isAdmin'];
+
+function RatingInputValue(props) {
+  const { item, applyValue, focusElementRef, headerFilterMenu } = props;
+
+  const ratingRef = React.useRef(null);
+  React.useImperativeHandle(focusElementRef, () => ({
+    focus: () => {
+      ratingRef.current
+        .querySelector(`input[value="${Number(item.value) || ''}"]`)
+        .focus();
+    },
+  }));
+
+  const handleFilterChange = (event, newValue) => {
+    applyValue({ ...item, value: newValue });
+  };
+
+  return (
+    <React.Fragment>
+      {headerFilterMenu}
+      <Box
+        sx={{
+          display: 'inline-flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          height: 20,
```

Comment:
```suggestion
          height: '100%',
```
To fix alignment in the filter panel